### ISSUE TITLE
Coreclr fails to build using source-build

### ISSUE THREAD

**Issue by jashook** [dotnet/coreclr#24888]
We specifically use .dotnet/dotnet instead of eng/dotnet.sh which does not correctly allow source build to override the dotnet to be used.

**Comment by jashook** [comment]
Example source-build output:

```
Commencing CoreCLR Repo build
Unknown CPU i386 detected, configuring as if for x64
__DistroRid: osx-x64
__RuntimeId: osx-x64
Setting up directories for build
Checking prerequisites...
Installing dotnet using Arcade...
/Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df//build.sh: line 161: /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/.dotnet/dotnet: No such file or directory
Failed to get PGO data package version.
```

**Comment by jashook** [comment]
After #24929 we hit more arcade issues:

```
/Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/common/msbuild.sh /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/empty.csproj /p:NativeVersionFile=/Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/bin/obj/OSX.x64.Release/version.c /p:ArcadeBuild=true /t:GenerateNativeVersionFile /restore /p:__BuildArch=x64 /p:__BuildType=Release /p:__BuildOS=OSX /p:RestoreDuringBuild=true /bl /p:ILLinkTrimAssembly=false /p:CheckCDefines=false /v:minimal /p:DotNetPackageVersionPropsPath=/Users/jashoo/projects/source-build/bin/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/Users/jashoo/projects/source-build/bin/obj/x64/Release/RestoreSources.props /p:DotNetOutputBlobFeedDir=/Users/jashoo/projects/source-build/bin/obj/x64/Release/blob-feed/
  /Users/jashoo/projects/source-build/Tools/dotnetcli/sdk/2.1.503/MSBuild.dll /nologo -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl /clp:Summary /nr:true /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=false /p:NativeVersionFile=/Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/bin/obj/OSX.x64.Release/version.c /p:ArcadeBuild=true /p:__BuildArch=x64 /p:__BuildType=Release /p:__BuildOS=OSX /p:RestoreDuringBuild=true /p:ILLinkTrimAssembly=false /p:CheckCDefines=false /p:DotNetPackageVersionPropsPath=/Users/jashoo/projects/source-build/bin/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/Users/jashoo/projects/source-build/bin/obj/x64/Release/RestoreSources.props /p:DotNetOutputBlobFeedDir=/Users/jashoo/projects/source-build/bin/obj/x64/Release/blob-feed/ /restore /t:GenerateNativeVersionFile /warnaserror /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/empty.csproj
    Restoring packages for /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/empty.csproj...
    Installing Microsoft.NETCore.DotNetAppHost 2.0.0.
    Installing Microsoft.NETCore.DotNetHostResolver 2.0.0.
    Installing Microsoft.NETCore.DotNetHostPolicy 2.0.0.
    Installing NETStandard.Library 2.0.0.
    Installing Microsoft.NETCore.Platforms 2.0.0.
    Installing Microsoft.NETCore.App 2.0.0.
    Installing Microsoft.Net.Compilers.Toolset 3.1.0-beta2-19067-14.
    Generating MSBuild file /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/artifacts/obj/empty/empty.csproj.nuget.g.props.
    Generating MSBuild file /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/artifacts/obj/empty/empty.csproj.nuget.g.targets.
    Restore completed in 1.78 sec for /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/empty.csproj.
  /Users/jashoo/projects/source-build/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19279.5/tools/Version.targets(51,11): error MSB4057: The target "InitializeSourceControlInformationFromSourceControlManager" does not exist in the project. [/Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/eng/empty.csproj]
```

**Comment by jashook** [comment]
The fix to above is to add skipgenerateversion to the coreclr build arguments in source-build. Opened https://github.com/dotnet/source-build/issues/1057 to address it.

**Comment by jashook** [comment]
New issue: with our move to `Microsoft.Build.Traversal` we have introduced a new pre-built into the build. Source-build fails with the following:

```
 Restoring packages for /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj...
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : error : [SourceBuiltSdkResolver] 0 overrides found for 'Microsoft.Build.Traversal'
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : warning : Unable to load the service index for source https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json.
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : warning : Unable to load the service index for source https://dotnet.myget.org/F/dotnet-core/api/v3/index.json.
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : warning : Unable to load the service index for source https://api.nuget.org/v3/index.json.
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : error : Unable to find package Microsoft.Build.Traversal. No packages exist with this id in source(s): dotnet-core, dotnet-coreclr, myget.org dotnet-core, nuget.org
  /Users/jashoo/projects/source-build/bin/src/coreclr.c6d6fc20a597b01ad0e2745b6933ca216746e7df/src/build.proj : error MSB4236: The SDK 'Microsoft.Build.Traversal' specified could not be found.
```
